util/ssh_client.go

```diff
@@ -29,6 +53,106 @@ func (c *sshClientImpl) ExecCommand(username string, password string, ip string,
 	return string(output), err
 }
 
+func (c *sshClientImpl) UploadFile(username string, password string, ip string, srcFile string, destFile string) error {
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+	}
+	if !IsIP(ip) {
+		return errors.New("invalid IP address")
+	}
+
+	if IsDir(srcFile) || IsDir(destFile) {
+		return errors.New("Is a directory")
+	}
+	client, err := ssh.Dial("tcp", ip+":22", config)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	conn := (*myssh.Client)(unsafe.Pointer(client))
```

Comment:
The sftp package require  golang.org/x/crypto/ssh type, but the ssh.Dial() return code.google.com/p/go.crypto/ssh type . I had to use unsafe.Pointer to convert type. 
And I had tried to use golang.org/x/crypto/ss package  to replace code.google.com/p/go.crypto/ssh package , the code can work but Travis CI will fails. 

So, I had to use unsafe.Pointer.